functions.inc.php

```diff
@@ -1013,6 +1013,32 @@ function pacrypt ($pw, $pw_db="") {
             $password = rtrim($password);
         }
     }
+	
+    elseif (substr($CONF['encrypt'], 0, 9) === 'php_crypt') {
+        // use PHPs crypt(), which uses the system's crypt()
+        // same algorithms as used in /etc/shadow
+        // you can have mixed hash types in the database for authentication, changed passwords get specified hash type
+        // the algorithm for a new hash is chosen by feeding a salt with correct magic to crypt()
+        // set $CONF['encrypt'] to 'php_crypt' to use the default MD5 crypt method
+        // set $CONF['encrypt'] to 'php_crypt:METHOD' to use another method; methods supported: DES, MD5, BLOWFISH, SHA256, SHA512
+        // tested on linux
+
+        if (strlen($pw_db) > 0) {
+            // existing pw provided. send entire password hash as salt for crypt() to figure out
+            $salt = $pw_db;
+        } else {
+            // no pw provided. create new password hash
+            if(strpos($CONF['encrypt'], ':') !== false) {
+				// use specified hash method
+                $split_method = preg_split ('/:/', $CONF['encrypt']);
```

Comment:
`explode(':', $CONF['encrypt'])` is probably faster ;-)

BTW: The comment line above has tabs instead of spaces.